### ISSUE TITLE
FFI type-safety lint should warn about fat pointers

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#20975]
```rust
extern {
    fn foo(x: *mut Fn());
}

fn main() {}
```

It's relatively unlikely that a fat pointer appearing in an FFI signature is correct, rather than just a typo/misunderstanding.

(Alternatively, one could just consider `[T]` and unsized traits to be non-FFI-safe.)

**Comment by steveklabnik** [comment]
This now gives

```
warning: found Rust trait type in foreign module, consider using a raw pointer instead, #[warn(improper_ctypes)] on by default
hello.rs:2         fn foo(x: *mut Fn());
```

:+1:
